### PR TITLE
Master dev

### PR DESCRIPTION
**Pull request by alexk2111**

### CODE REVIEW

File: StartAndroid/app/src/main/java/com/sigmasoftwere/akucherenko/startandroid/MainActivity.kt [hunk 1]
```diff
@@ -2,10 +2,76 @@ package com.sigmasoftwere.akucherenko.startandroid
...
+import android.widget.TextView
+import android.widget.Toast
+import androidx.core.content.ContextCompat
+
+private const val KEY_COUNT = "count"
 
 class MainActivity : AppCompatActivity() {
+
+    private val LOG_TAG = MainActivity::class.java.simpleName
+    private var mCount = 0
```
Comment (ASKabanets) [inline StartAndroid/app/src/main/java/com/sigmasoftwere/akucherenko/startandroid/MainActivity.kt]: `mCount ` -> `count `
Same with others

File: StartAndroid/app/src/main/java/com/sigmasoftwere/akucherenko/startandroid/MainActivity.kt [hunk 2]
```diff
@@ -2,10 +2,76 @@ package com.sigmasoftwere.akucherenko.startandroid
...
+import android.widget.Button
+import android.widget.TextView
+import android.widget.Toast
+import androidx.core.content.ContextCompat
+
+private const val KEY_COUNT = "count"
 
 class MainActivity : AppCompatActivity() {
+
+    private val LOG_TAG = MainActivity::class.java.simpleName
```
Comment (ASKabanets) [inline StartAndroid/app/src/main/java/com/sigmasoftwere/akucherenko/startandroid/MainActivity.kt]: Ok, but try to use https://github.com/JakeWharton/timber
It's kinda "industry standard"

File: StartAndroid/app/src/main/java/com/sigmasoftwere/akucherenko/startandroid/MainActivity.kt [hunk 3]
```diff
@@ -2,10 +2,76 @@ package com.sigmasoftwere.akucherenko.startandroid
...
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+
+        Log.d(LOG_TAG, "Hello World")
+
+        mCount = savedInstanceState?.getInt(KEY_COUNT, 0)
+            ?: getString(R.string.count_initial_value).toInt()
+
+        mShowCount = findViewById(R.id.tv_show_count)
```
Comment (ASKabanets) [inline StartAndroid/app/src/main/java/com/sigmasoftwere/akucherenko/startandroid/MainActivity.kt]: Try to use https://medium.com/androiddevelopers/use-view-binding-to-replace-findviewbyid-c83942471fc instead `findViewById`

File: StartAndroid/app/src/main/java/com/sigmasoftwere/akucherenko/startandroid/MainActivity.kt [hunk 4]
```diff
@@ -2,10 +2,76 @@ package com.sigmasoftwere.akucherenko.startandroid
...
+        mShowCount.text = mCount.toString()
+
+        mReset = findViewById(R.id.button_reset)
+        setResetColor()
+
+//        mCount = getString(R.string.count_initial_value).toInt()
+    }
+
+    fun showToast(view: android.view.View) {
+        Toast.makeText(this, R.string.toast_message, Toast.LENGTH_SHORT).show()
```
Comment (ASKabanets) [inline StartAndroid/app/src/main/java/com/sigmasoftwere/akucherenko/startandroid/MainActivity.kt]: No reason to make single-line function for showing toast

File: StartAndroid/app/src/main/java/com/sigmasoftwere/akucherenko/startandroid/MainActivity.kt [hunk 5]
```diff
@@ -2,10 +2,76 @@ package com.sigmasoftwere.akucherenko.startandroid
...
+    }
+
+    fun resetCount(view: android.view.View) {
+        mCount = 0
+        mShowCount.text = mCount.toString()
+        setResetColor()
+    }
+
+    fun setResetColor() {
+        if (mCount > 0)
```
Comment (ASKabanets) [inline StartAndroid/app/src/main/java/com/sigmasoftwere/akucherenko/startandroid/MainActivity.kt]: Add brackets

File: StartAndroid/app/src/main/res/layout-land/activity_main.xml [hunk 6]
```diff
@@ -0,0 +1,69 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".MainActivity">
+
+    <TextView
+        android:id="@+id/tv_show_count"
```
Comment (ASKabanets) [inline StartAndroid/app/src/main/res/layout-land/activity_main.xml]: `tv_show_count ` -> `showCountTextView`
Similar with others

File: StartAndroid/app/src/main/res/layout-land/activity_main.xml [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+        android:layout_width="0dp"
+        android:layout_height="0dp"
+        android:background="#FFFF00"
```
Comment (ASKabanets) [inline StartAndroid/app/src/main/res/layout-land/activity_main.xml]: `#FFFF00` move to `colors.xml` resources

File: TwoActivities/app/src/main/java/com/sigmasoftwere/akucherenko/twoactivities/MainActivity.kt [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+import android.os.Bundle
+import android.util.Log
+import android.content.Intent
+import android.widget.EditText
+
+
+const val EXTRA_MESSAGE = "com.sigmasoftwere.akucherenko.twoactivities.extra.MESSAGE"
+class MainActivity : AppCompatActivity() {
+    private val LOG_TAG = MainActivity::class.java.simpleName
+    private lateinit var mMessageEditText: EditText
```
Comment (ASKabanets) [inline TwoActivities/app/src/main/java/com/sigmasoftwere/akucherenko/twoactivities/MainActivity.kt]: `mMessageEditText ` -> `messageEditText `

File: TwoActivities/app/src/main/java/com/sigmasoftwere/akucherenko/twoactivities/MainActivity.kt [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.util.Log
+import android.content.Intent
+import android.widget.EditText
+
+
+const val EXTRA_MESSAGE = "com.sigmasoftwere.akucherenko.twoactivities.extra.MESSAGE"
+class MainActivity : AppCompatActivity() {
```
Comment (ASKabanets) [inline TwoActivities/app/src/main/java/com/sigmasoftwere/akucherenko/twoactivities/MainActivity.kt:12]: Add empty line

File: TwoActivities/app/src/main/java/com/sigmasoftwere/akucherenko/twoactivities/MainActivity.kt [hunk 10]
```diff
@@ -0,0 +1,29 @@
+package com.sigmasoftwere.akucherenko.twoactivities
+
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.util.Log
+import android.content.Intent
+import android.widget.EditText
+
+
```
Comment (ASKabanets) [inline TwoActivities/app/src/main/java/com/sigmasoftwere/akucherenko/twoactivities/MainActivity.kt:9]: Remove empty line

File: TwoActivities/app/src/main/java/com/sigmasoftwere/akucherenko/twoactivities/SecondActivity.kt [hunk 11]
```diff
@@ -0,0 +1,33 @@
+package com.sigmasoftwere.akucherenko.twoactivities
+
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.widget.EditText
+import android.widget.TextView
+
+const val  EXTRA_REPLY = "com.sigmasoftwere.akucherenko.twoactivities.extra.REPLY"
```
Comment (ASKabanets) [inline TwoActivities/app/src/main/java/com/sigmasoftwere/akucherenko/twoactivities/SecondActivity.kt]: Remove space

File: TwoActivities/app/src/main/java/com/sigmasoftwere/akucherenko/twoactivities/SecondActivity.kt [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+
+class SecondActivity : AppCompatActivity() {
+
+    private lateinit var messageText: TextView
+    private lateinit var mReply: EditText
```
Comment (ASKabanets) [inline TwoActivities/app/src/main/java/com/sigmasoftwere/akucherenko/twoactivities/SecondActivity.kt]: Naming

File: TwoActivities/app/src/main/java/com/sigmasoftwere/akucherenko/twoactivities/SecondActivity.kt [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+    override fun onCreate(savedInstanceState: Bundle?) {
```
Comment (ASKabanets) [inline TwoActivities/app/src/main/java/com/sigmasoftwere/akucherenko/twoactivities/SecondActivity.kt:18]: Add empty line

File: TwoActivities/app/src/main/res/layout/activity_main.xml [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+    tools:context=".MainActivity">
+
+    <LinearLayout
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:gravity="end"
+        android:orientation="horizontal">
+
+        <EditText
+            android:id="@+id/editText_main"
```
Comment (ASKabanets) [inline TwoActivities/app/src/main/res/layout/activity_main.xml]: Id's naming

File: TwoActivities/app/src/main/res/layout/activity_second.xml [hunk 15]
```diff
@@ -0,0 +1,58 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:orientation="vertical"
+    tools:context=".SecondActivity">
+
+    <TextView
+        android:id="@+id/text_header"
```
Comment (ASKabanets) [inline TwoActivities/app/src/main/res/layout/activity_second.xml]: Id's naming

File: StartAndroid/.idea/.gitignore [hunk 16]
```diff
@@ -0,0 +1,3 @@
+# Default ignored files
```
Comment (ASKabanets) [inline StartAndroid/.idea/.gitignore:1]: Add `.idea` folder to `.gitignore`

**Review by ASKabanets — CHANGES_REQUESTED** [review]
OK in general. Minor comments. Also, branch name and PR name should be more self-explained than `master-dev`

File: StartAndroid/app/src/main/java/com/sigmasoftwere/akucherenko/startandroid/MainActivity.kt (hunk 1 above)
Comment (alexk2111) [reply StartAndroid/app/src/main/java/com/sigmasoftwere/akucherenko/startandroid/MainActivity.kt]: Variable renamed

File: StartAndroid/app/src/main/java/com/sigmasoftwere/akucherenko/startandroid/MainActivity.kt (hunk 2 above)
Comment (alexk2111) [reply StartAndroid/app/src/main/java/com/sigmasoftwere/akucherenko/startandroid/MainActivity.kt]: Done

File: StartAndroid/app/src/main/java/com/sigmasoftwere/akucherenko/startandroid/MainActivity.kt (hunk 3 above)
Comment (alexk2111) [reply StartAndroid/app/src/main/java/com/sigmasoftwere/akucherenko/startandroid/MainActivity.kt]: Done

File: StartAndroid/app/src/main/java/com/sigmasoftwere/akucherenko/startandroid/MainActivity.kt (hunk 4 above)
Comment (alexk2111) [reply StartAndroid/app/src/main/java/com/sigmasoftwere/akucherenko/startandroid/MainActivity.kt]: Replaced Toast in clickListener

File: StartAndroid/app/src/main/java/com/sigmasoftwere/akucherenko/startandroid/MainActivity.kt (hunk 5 above)
Comment (alexk2111) [reply StartAndroid/app/src/main/java/com/sigmasoftwere/akucherenko/startandroid/MainActivity.kt]: Done

File: StartAndroid/app/src/main/res/layout-land/activity_main.xml (hunk 6 above)
Comment (alexk2111) [reply StartAndroid/app/src/main/res/layout-land/activity_main.xml]: All id's renamed

File: StartAndroid/app/src/main/res/layout-land/activity_main.xml (hunk 7 above)
Comment (alexk2111) [reply StartAndroid/app/src/main/res/layout-land/activity_main.xml]: Color moved to color.xml

File: TwoActivities/app/src/main/java/com/sigmasoftwere/akucherenko/twoactivities/MainActivity.kt (hunk 8 above)
Comment (alexk2111) [reply TwoActivities/app/src/main/java/com/sigmasoftwere/akucherenko/twoactivities/MainActivity.kt]: Variables renamed

File: TwoActivities/app/src/main/java/com/sigmasoftwere/akucherenko/twoactivities/MainActivity.kt (hunk 9 above)
Comment (alexk2111) [reply TwoActivities/app/src/main/java/com/sigmasoftwere/akucherenko/twoactivities/MainActivity.kt:12]: line added

File: TwoActivities/app/src/main/java/com/sigmasoftwere/akucherenko/twoactivities/MainActivity.kt (hunk 10 above)
Comment (alexk2111) [reply TwoActivities/app/src/main/java/com/sigmasoftwere/akucherenko/twoactivities/MainActivity.kt:9]: line removed

File: TwoActivities/app/src/main/java/com/sigmasoftwere/akucherenko/twoactivities/SecondActivity.kt (hunk 11 above)
Comment (alexk2111) [reply TwoActivities/app/src/main/java/com/sigmasoftwere/akucherenko/twoactivities/SecondActivity.kt]: Space removed

File: TwoActivities/app/src/main/java/com/sigmasoftwere/akucherenko/twoactivities/SecondActivity.kt (hunk 12 above)
Comment (alexk2111) [reply TwoActivities/app/src/main/java/com/sigmasoftwere/akucherenko/twoactivities/SecondActivity.kt]: Renamed

File: TwoActivities/app/src/main/java/com/sigmasoftwere/akucherenko/twoactivities/SecondActivity.kt (hunk 13 above)
Comment (alexk2111) [reply TwoActivities/app/src/main/java/com/sigmasoftwere/akucherenko/twoactivities/SecondActivity.kt:18]: Line added

File: TwoActivities/app/src/main/res/layout/activity_main.xml (hunk 14 above)
Comment (alexk2111) [reply TwoActivities/app/src/main/res/layout/activity_main.xml]: Id's renamed

File: StartAndroid/.idea/.gitignore (hunk 16 above)
Comment (alexk2111) [reply StartAndroid/.idea/.gitignore:1]: Folder `.idea` added to `.gitignore`

File: TwoActivities/app/src/main/res/layout/activity_second.xml (hunk 15 above)
Comment (alexk2111) [reply TwoActivities/app/src/main/res/layout/activity_second.xml]: Id's renamed

**Pull request merged**